### PR TITLE
Adjust the relative source file path when indexing sources in a multi…

### PR DESCRIPTION
**Pull request by MarkKharitonov**
…-repo checkout build.

**Task name**: PublishSymbolsV2

**Description**: Adjusts the relative path of the indexed source files when multi-repo checkout is used

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** https://github.com/microsoft/azure-pipelines-tasks/issues/14852

**Checklist**:
- [X] Task version was bumped - done.
- [X] Tested the code changes offline, since I have no idea how to test an Azure DevOps task.

### CODE REVIEW

**Comment by MarkKharitonov** [comment]
Anyone?

**Comment by weikanglim** [comment]
Hey Mark, thanks for submitting this change. We would need to  test this in a full build end-to-end with the scenario in mind, and ensure that the new symbols are indexed correctly. I'm a little confused if the `TFS_REPO` would even be correct in the multi-repo scenario. Fixing the relative source directory is one thing, but the repository might still be incorrect.

**Comment by MarkKharitonov** [comment]
Could you comment on the code itself, so that the context of the comment is clear?

File: Tasks/PublishSymbolsV2/IndexHelpers/SrcSrvIniContentFunctions.ps1 [hunk 1]
```diff
@@ -82,8 +82,18 @@ function New-TfsGitSrcSrvIniContent {
...
+                if ($i -gt 0) {
+                    $repoName = $relativeSourceFilePath.Substring(0, $i)
+                    if (Test-Path "${sourcesRootPath}$repoName\.git") {
+                        $relativeSourceFilePath = $relativeSourceFilePath.Substring($i + 1)
+                    }
+                }
+            }
             $relativeSourceFilePath = $relativeSourceFilePath.Replace('\', '/')
             $relativeSourceFilePath = "/$relativeSourceFilePath"
             "$sourceFilePath*TFS_COLLECTION*TFS_TEAM_PROJECT*TFS_REPO*TFS_COMMIT*TFS_SHORT_COMMIT*$relativeSourceFilePath*TFS_APPLY_FILTERS"
```
[the remark this answers is not in the document]
Comment (carl-tanner) [reply Tasks/PublishSymbolsV2/IndexHelpers/SrcSrvIniContentFunctions.ps1:99]: I've had a chance this week to test it locally and validated that for the "self" repo being checked out information is now correct and for others the repo info is incorrect.  I'll get this PR approved.

**Review by fadnavistanmay — APPROVED** [review]

File: Tasks/PublishSymbolsV2/IndexHelpers/SrcSrvIniContentFunctions.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarkKharitonov) [reply Tasks/PublishSymbolsV2/IndexHelpers/SrcSrvIniContentFunctions.ps1:99]: Thank you so much.

**Pull request merged**